examples/fp8_attention.py

```diff
@@ -0,0 +1,233 @@
+                qk_shifted = qk_scaled - m_new[:, None]
+
+                # Use exp2 to match Triton kernel's implementation
+                # Note: Triton kernel already multiplies sm_scale by 1.44269504
+                p = torch.exp2(qk_shifted)  # [tile_m, tile_n]
+
+                # Sum of exponentials for this block
+                l_ij = torch.sum(p, dim=-1)  # [tile_m]
+
+                # Update accumulators with correction factor
+                # Correction factor for previous blocks
+                alpha = torch.exp2(m_i - m_new)
+                l_i = l_i * alpha + l_ij
+                acc = acc * alpha[:, None]
+
+                # Load values - V is [dim, seq]
+                v_tile = v[bh, :, tile_n]  # [dim, tile_n] - keep in FP8
+
+                # Convert p to FP8 for FP8 GEMM
+                p_fp8 = p.to(v.dtype)  # Convert to same FP8 type as V
+
+                # Accumulate attention @ V with FP8 GEMM
+                v_t = v_tile.transpose(0, 1)  # [tile_n, dim]
+                pv = torch.matmul(p_fp8, v_t).to(torch.float32)  # [tile_m, dim]
+                acc = acc + pv
+
+                # Update max tracker
+                m_i = m_new
+
+            # Final normalization
+            acc = acc / l_i[:, None]
+            out[bh, tile_m, :] = acc
+
+    return out
+
+
+def prepare_fp8_attention_inputs(
+    q: torch.Tensor,  # [batch, heads, seq, dim]
+    k: torch.Tensor,  # [batch, heads, seq, dim]
+    v: torch.Tensor,  # [batch, heads, seq, dim]
+) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor, tuple[int, int, int, int]]:
+    """
+    Common preprocessing for FP8 attention implementations.
+
+    Returns:
+        q_reshaped_fp8: [batch*heads, seq, dim] - in FP8 e5m2
+        k_reshaped_fp8: [batch*heads, seq, dim] - in FP8 e5m2
+        v_transposed_fp8: [batch*heads, dim, seq] - in FP8 e5m2
+        shape: (batch, heads, seq_len, head_dim)
+    """
+    batch, heads, seq_len, head_dim = q.shape
+
+    # Reshape to [batch*heads, seq, dim]
+    q_reshaped = q.reshape(batch * heads, seq_len, head_dim)
+    k_reshaped = k.reshape(batch * heads, seq_len, head_dim)
+
+    # Transpose V to [batch, heads, dim, seq] then reshape
+    v_transposed = v.permute(0, 1, 3, 2).reshape(batch * heads, head_dim, seq_len)
+
+    # Convert to FP8 e5m2
+    q_reshaped_fp8 = q_reshaped.to(torch.float8_e5m2)
+    k_reshaped_fp8 = k_reshaped.to(torch.float8_e5m2)
+    v_transposed_fp8 = v_transposed.to(torch.float8_e5m2)
+
+    return (
+        q_reshaped_fp8,
+        k_reshaped_fp8,
+        v_transposed_fp8,
+        (batch, heads, seq_len, head_dim),
+    )
+
+
+def fp8_attention_tritonbench(
+    q: torch.Tensor, k: torch.Tensor, v: torch.Tensor
+) -> torch.Tensor:
+    """Wrapper for TritonBench compatibility."""
+    # Common preprocessing with FP8 conversion
+    q_fp8, k_fp8, v_fp8, shape = prepare_fp8_attention_inputs(q, k, v)
+    batch, heads, seq_len, head_dim = shape
```

Comment:
Do we want to benchmark this?  The PyTorch input prepare ins't really helion work.

Reply:
agreed - excluded the input preprocessing part from timing measurement (the Triton kernels in tritonbench also have this excluded)